app/services/interview_workflow_constraints.rb

```diff
@@ -0,0 +1,59 @@
+class InterviewWorkflowConstraints
+  STATES_ALLOWING_INTERVIEW_CHANGES = ApplicationStateChange::INTERVIEWABLE_STATES.map(&:to_s).freeze
+
+  attr_reader :interview, :today
+  delegate :application_choice, to: :interview
+
+  def initialize(interview:)
+    @interview = interview
+    @today = Time.zone.now.beginning_of_day
+  end
+
+  def create!
+    application_not_in_interviewing_states?
+  end
+
+  def update!
+    raise_error_if_interview_has_no_changes
+
+    interview_in_the_past?
+    interview_already_cancelled?
+    application_not_in_interviewing_states?
+  end
+
+  def cancel!
+    interview_in_the_past?
+    interview_already_cancelled?
+    application_not_in_interviewing_states?
+  end
+
+  def raise_error_if_interview_has_no_changes
```

Comment:
Could we remove this as this will also break manage, as it deviates from our standard update in place (similar with make offer)?